Formula/blis.rb

```diff
@@ -1,19 +1,52 @@
 class Blis < Formula
   desc "BLAS-like Library Instantiation Software Framework"
   homepage "https://github.com/flame/blis"
-  url "https://github.com/flame/blis/archive/0.8.1.tar.gz"
-  sha256 "729694128719801e82fae7b5f2489ab73e4a467f46271beff09588c9265a697b"
   license "BSD-3-Clause"
   head "https://github.com/flame/blis.git"
 
+  stable do
+    url "https://github.com/flame/blis/archive/0.8.1.tar.gz"
+    sha256 "729694128719801e82fae7b5f2489ab73e4a467f46271beff09588c9265a697b"
+
+    # Fix non-generic build for Apple ARM and compilation with Clang.
+    # Upstream ref: https://github.com/flame/blis/pull/506
+    # Remove in the next release.
+    patch do
+      url "https://github.com/flame/blis/commit/bf727636632a368f3247dc8ab1d4b6119e9c511a.patch?full_index=1"
+      sha256 "55962e5df20308c08a5b10a2b7186268dfc905127e360af17e271198d2cde687"
+    end
+
+    # Allow using `thunderx2` config on Apple ARM with Clang
+    # Upstream ref: https://github.com/flame/blis/pull/492
+    # Remove in the next release.
+    patch do
+      url "https://github.com/flame/blis/commit/6548cebaf55a1f9bdb8417cc89dd0444d8f9c2e4.patch?full_index=1"
+      sha256 "c7f2ae70662b33f8db0aeb2773467288a9fba03fd26005d71d9ac0b9ff712c60"
+    end
```

Comment:
This patch is a no-op. `brew` [filters out `-mcpu` flags](https://github.com/Homebrew/brew/blob/bdc3391a1319f1c40a108d14a077ca0964b33a3f/Library/Homebrew/shims/super/cc#L176).

Reply:
Without patch, the build automatically fails since it has hard restriction on GCC. So, would need to add GCC dependency to get ARM to build.

Not sure of exact behavior of resulting libraries though. It is possible that result is sub-optimal, but this would require benchmarking to compare with baseline `generic` variant.

Reply:
Does the `thunderx2` config differ from the `generic` one more than in the `-mcpu` flag?

Reply:
It has some kernel block size and other arch-specific settings, but these are optimized for ThunderX2 rather than M1, so hard to say if "better":
- ThunderX2: https://github.com/flame/blis/blob/master/config/thunderx2/bli_cntx_init_thunderx2.c
- Generic: https://github.com/flame/blis/blob/master/config/generic/bli_cntx_init_generic.c

Reply:
I don't know best way to benchmark this, but using the testsuite, there are some tests that perform identically, while others that ThunderX2 is an order of magnitude more performant than Generic (e.g. 80 gflops vs 5 gflops), so I think we should at minimum compile with ThunderX2 profile.

It is possible that compiling with GCC+OpenMP may perform better, but don't want to experiment with all variations.

ThunderX2 compiled via Homebrew:
```
% blis_<dt><op>_<params>_<stor>      m     n     k   gflops   resid      result
blis_sgemm_nn_rrr                  100   100   100    19.50   1.59e-08   PASS
blis_sgemm_nn_rrr                  200   200   200    76.71   1.06e-08   PASS
blis_sgemm_nn_rrr                  300   300   300    81.95   1.00e-08   PASS
blis_sgemm_nn_rrr                  400   400   400    86.74   1.58e-08   PASS
blis_sgemm_nn_rrr                  500   500   500    86.89   1.12e-08   PASS
```

Generic compiled via Homebrew:
```
% blis_<dt><op>_<params>_<stor>      m     n     k   gflops   resid      result
blis_sgemm_nn_rrr                  100   100   100     4.12   1.70e-08   PASS
blis_sgemm_nn_rrr                  200   200   200     5.16   1.05e-08   PASS
blis_sgemm_nn_rrr                  300   300   300     5.29   1.02e-08   PASS
blis_sgemm_nn_rrr                  400   400   400     5.38   1.33e-08   PASS
blis_sgemm_nn_rrr                  500   500   500     5.30   1.08e-08   PASS
```